deployability/modules/testing/README.MD

```diff
@@ -0,0 +1,341 @@
+        - dependencies:
+          - manager: "{working-dir}/manager-{manager-os}/inventory.yaml"
+          - agent: "{working-dir}/agent-{agent}/inventory.yaml"
+        - tests: "install,basic_info,connection,register,repo,restart,stop,uninstall"
+        - component: "agent"
+        - wazuh-version: "4.7.2"
+        - wazuh-revision: "1"
+        - live: False
+```
+
+In the exposed fixture fragment, it can be observed that for the execution of the testing module launcher (`testing/main.py`), it is necessary to provide the inventory, dependencies, component, tests to execute, Wazuh version, Wazuh revision, and whether the repository is live or not (if not, it will look for information in `packages-dev` pre-release).
+
+For manual execution, an example command would be:
+
+```bash
+python3 modules/testing/main.py --inventory "/tmp/dtt1-poc/agent-linux-ubuntu-22.04-amd64/inventory.yaml" --component "agent" --wazuh-version "4.7.2" --wazuh-revision "1" --tests install --dependencies '{"manager":"/tmp/dtt1-poc/manager-linux-ubuntu-22.04-amd64/inventory.yaml"}' --live False
+```
+
+The module is composed of:
+
+- **Launcher** (`/wazuh-qa/deployability/modules/testing/main.py`): Entry point for the workflow or the user who wishes to execute a test.
+  
+- **Parameter validator** (`/wazuh-qa/deployability/modules/testing/models.py`): Validator for parameters entered by the Workflow or the user.
+  
+- **Module functions** (`/wazuh-qa/deployability/modules/testing/testing.py`): Module-specific functions responsible for triggering the test.
+  
+- **Executor** (`/wazuh-qa/deployability/modules/provision/actions.py`): Executes the provision using Actions in the generic module.
+  
+- **Playbooks** (`/wazuh-qa/deployability/modules/playbooks`): Contains setup, execution, and post-test cleanup playbooks.
+  
+- **Tests** (`/wazuh-qa/deployability/modules/tests`): Contains the test_agent and test_manager directories where the tests to be executed are located.
+  
+- **Helpers** (`/wazuh-qa/deployability/modules/tests/helpers`): Contains files that enable the execution of the test.
+  
+- **Conftest** (`/wazuh-qa/deployability/modules/tests/conftest.py`): Contains information that will be transferred from the parameter validator to the tests.
+
+
+![image](https://github.com/wazuh/wazuh-qa/assets/125690423/58332378-489a-49da-a73a-4d87d6bed3bc)
+
+
+The testing module should receive the infrastructure generated and provisioned by the allocation and provision modules. The module has the ability to execute actions on the hosts as well as perform the necessary validation.
+
+In the case of agent tests, the provisioning of the manager must be done by the provisioner. The install test will perform both installation and validation, so it should always be placed as the first test in the test list. The same situation will occur with the uninstall test, which will go to the end of the test list because it will execute the uninstallation prior to validation.
```

Comment:
prior to -> before

Reply:
Done https://github.com/wazuh/wazuh-qa/pull/5005/commits/66983858545c260b04b271ca0cd647627fc80d08